vertx-web/src/main/java/io/vertx/ext/web/handler/StaticHandler.java

```diff
@@ -234,6 +235,15 @@ static StaticHandler create(String root, ClassLoader classLoader) {
   StaticHandler setMaxCacheSize(int maxCacheSize);
 
   /**
+   * Set the file mapping for http2push and link preload
+   *
+   * @param http2PushMapping the mapping for http2 push
+   * @return a reference to this, so the API can be used fluently
+   */
+  @Fluent
+  StaticHandler setHttp2PushMapping(JsonObject http2PushMapping);
```

Comment:
I think that we should replace this mapping with a data object. in this case we could have a list with a object that would contain:

```json
{
  href: "/app/file",
  as: "style/script/etc...",
  nopush: false
}
```

Reply:
It would be an array of these data objects (for say index.html file as key) I presume?

Reply:
Yes